micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java

```diff
@@ -31,9 +31,17 @@ public class MicrometerMetrics implements Metrics {
   private static final String RECONCILIATIONS_QUEUE_SIZE = PREFIX + RECONCILIATIONS + "queue.size.";
   private final MeterRegistry registry;
   private final Map<String, AtomicInteger> gauges = new ConcurrentHashMap<>();
+  private final Map<ResourceID, Set<Meter.Id>> metersPerResource = new ConcurrentHashMap<>();
+  private final ScheduledExecutorService metersCleaner = Executors.newScheduledThreadPool(10);
+  private final int cleanUpDelayInSeconds;
 
   public MicrometerMetrics(MeterRegistry registry) {
+    this(registry, 300);
+  }
+
+  public MicrometerMetrics(MeterRegistry registry, int cleanUpDelayInSeconds) {
     this.registry = registry;
+    this.cleanUpDelayInSeconds = cleanUpDelayInSeconds;
```

Comment:
Why is there a delay, why cannot be deleted instantly?

Reply:
Because you still might want to get the metrics on the deleted resources for a while before the metrics are removed.

Reply:
true, would worth to write it as a javadoc so it is obvious from code.

Reply:
Yes, the feature needs to be documented.